server/client_manager.py

```diff
@@ -324,4 +331,28 @@ def get_ooc_muted_clients(self):
         for client in self.clients:
             if client.is_ooc_muted:
                 clients.append(client)
-        return clients
\ No newline at end of file
+        return clients
+
+    def get_hdid_by_ip(self, ip):
+        for client in self.clients:
+            hdid = client.get_hdid()
+            for client in self.clients:
+                if client.get_hdid() is not None and client.get_ip() == ip:
+                    hdid = client.get_hdid()
+                    return hdid
+        if hdid == None:
+            return None
+        else:
+            return hdid
```

Comment:
I can't confess to knowing exactly what that would look like.

`if hdid == None
    return hdid or None`

???